packages/flutter/lib/src/material/button_style.dart

```diff
@@ -120,58 +120,58 @@ class ButtonStyle with Diagnosticable {
   ///
   /// The color of the [textStyle] is typically not used directly, the
   /// [foregroundColor] is used instead.
-  final MaterialStateProperty<TextStyle>? textStyle;
+  final MaterialStateProperty<TextStyle?>? textStyle;
```

Comment:
@HansMuller this double-level of nullness looks suspicious, is it what we want?

Reply:
Yes, this is correct. If specified (if non-null), properties like textStyle can resolve to `TextStyle?`.  If the MSP is null,  then the style does not override the default textStyle. If the MSP is non-null, it can resolve to null for the subset of MaterialStates that it does not override.